expo에서의 android install 문제

### Description

안드로이드에서 의존성 설치가 안되서 build가 실패하는 것 같습니다.

```
> expo run:android

(node:73249) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
› Building app...
Starting a Gradle Daemon (subsequent builds will be faster)
Configuration on demand is an incubating feature.

> Configure project :expo

Using expo modules
  - expo-constants (15.4.5)
  - expo-file-system (16.0.5)
  - expo-font (11.10.2)
  - expo-image (1.10.6)
  - expo-keep-awake (12.8.2)
  - expo-modules-core$android-annotation-processor (1.11.8)
  - expo-modules-core (1.11.8)
  - expo-modules-core$android-annotation (1.11.8)
  - expo-splash-screen (0.26.4)
  - expo-system-ui (2.9.3)
  - expo-web-browser (12.8.2)


FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':app:processDebugResources'.
> Could not resolve all task dependencies for configuration ':app:debugRuntimeClasspath'.
   > Could not find com.kakao.sdk:v2-user:2.11.2.
     Searched in the following locations:
       - https://oss.sonatype.org/content/repositories/snapshots/com/kakao/sdk/v2-user/2.11.2/v2-user-2.11.2.pom
       - https://repo.maven.apache.org/maven2/com/kakao/sdk/v2-user/2.11.2/v2-user-2.11.2.pom
       - file:/Users/user/projects/myapp/node_modules/jsc-android/dist/com/kakao/sdk/v2-user/2.11.2/v2-user-2.11.2.pom
       - https://dl.google.com/dl/android/maven2/com/kakao/sdk/v2-user/2.11.2/v2-user-2.11.2.pom
       - https://www.jitpack.io/com/kakao/sdk/v2-user/2.11.2/v2-user-2.11.2.pom
       - file:/Users/user/projects/myapp/node_modules/react-native/android/com/kakao/sdk/v2-user/2.11.2/v2-user-2.11.2.pom
     Required by:
         project :app > project :react-native-seoul_kakao-login
   > Could not find com.kakao.sdk:v2-talk:2.11.2.
     Searched in the following locations:
       - https://oss.sonatype.org/content/repositories/snapshots/com/kakao/sdk/v2-talk/2.11.2/v2-talk-2.11.2.pom
       - https://repo.maven.apache.org/maven2/com/kakao/sdk/v2-talk/2.11.2/v2-talk-2.11.2.pom
       - file:/Users/user/projects/myapp/node_modules/jsc-android/dist/com/kakao/sdk/v2-talk/2.11.2/v2-talk-2.11.2.pom
       - https://dl.google.com/dl/android/maven2/com/kakao/sdk/v2-talk/2.11.2/v2-talk-2.11.2.pom
       - https://www.jitpack.io/com/kakao/sdk/v2-talk/2.11.2/v2-talk-2.11.2.pom
       - file:/Users/user/projects/myapp/node_modules/react-native/android/com/kakao/sdk/v2-talk/2.11.2/v2-talk-2.11.2.pom
     Required by:
         project :app > project :react-native-seoul_kakao-login
   > Could not find com.kakao.sdk:v2-story:2.11.2.
     Searched in the following locations:
       - https://oss.sonatype.org/content/repositories/snapshots/com/kakao/sdk/v2-story/2.11.2/v2-story-2.11.2.pom
       - https://repo.maven.apache.org/maven2/com/kakao/sdk/v2-story/2.11.2/v2-story-2.11.2.pom
       - file:/Users/user/projects/myapp/node_modules/jsc-android/dist/com/kakao/sdk/v2-story/2.11.2/v2-story-2.11.2.pom
       - https://dl.google.com/dl/android/maven2/com/kakao/sdk/v2-story/2.11.2/v2-story-2.11.2.pom
       - https://www.jitpack.io/com/kakao/sdk/v2-story/2.11.2/v2-story-2.11.2.pom
       - file:/Users/user/projects/myapp/node_modules/react-native/android/com/kakao/sdk/v2-story/2.11.2/v2-story-2.11.2.pom
     Required by:
         project :app > project :react-native-seoul_kakao-login

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 24s
5 actionable tasks: 5 up-to-date
Error: /Users/user/projects/myapp/android/gradlew exited with non-zero code: 1
Error: /Users/user/projects/myapp/android/gradlew exited with non-zero code: 1
    at ChildProcess.completionListener (/Users/user/projects/myapp/node_modules/@expo/spawn-async/build/spawnAsync.js:52:23)
    at Object.onceWrapper (node:events:634:26)
    at ChildProcess.emit (node:events:519:28)
    at maybeClose (node:internal/child_process:1105:16)
    at ChildProcess._handle.onexit (node:internal/child_process:305:5)
    ...
    at Object.spawnAsync [as default] (/Users/user/projects/myapp/node_modules/@expo/spawn-async/build/spawnAsync.js:17:21)
    at spawnGradleAsync (/Users/user/projects/myapp/node_modules/@expo/cli/build/src/start/platforms/android/gradle.js:72:46)
    at Object.assembleAsync (/Users/user/projects/myapp/node_modules/@expo/cli/build/src/start/platforms/android/gradle.js:52:18)
    at runAndroidAsync (/Users/user/projects/myapp/node_modules/@expo/cli/build/src/run/android/runAndroidAsync.js:36:24)

```

카카오 개발자 페이지에서 설정은 README에 나온 대로 모두 해두었습니다.
다만 README '설치' 부분에서 npm install 하는 부분밖에 안나와 있어서 android, ios에서의 설치는 어떻게 되는지 잘 모르겠습니다. autolinking이 된다는 게 native쪽에도 설치된다 이 말씀이시죠? 참고로 react-native 0.73.2 버전을 사용하고 있습니다.

build.gradle에 수동으로 아래와 같이 설치해줘서 다시 빌드했더니 이번엔 잘 되긴 합니다만 이게 올바른 방법인지 모르겠어서 여쭤봅니다. ㅠㅠ 앱 개발은 처음이라... 양해부탁드립니다.

```
allprojects {
    repositories {
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url(new File(['node', '--print', "require.resolve('react-native/package.json')"].execute(null, rootDir).text.trim(), '../android'))
        }
        // 이 부분 추가
        maven { url 'https://devrepo.kakao.com/nexus/content/groups/public/' }
        maven {
            // Android JSC is installed from npm
            url(new File(['node', '--print', "require.resolve('jsc-android/package.json', { paths: [require.resolve('react-native/package.json')] })"].execute(null, rootDir).text.trim(), '../dist'))
        }

        google()
        mavenCentral()
        maven { url 'https://www.jitpack.io' }
    }
}

```


### Steps to reproduce

1. expo prebuild
2. npm install @react-native-seoul/kakao-login
3. npm run android(= `expo run:android`)

따로 eas build는 사용하고 있지 않습니다.

### Snack or a link to a repository

-

### React Native Kakao Login version

5.3.1

### React Native version

0.73.2

### Platforms

Android

### JavaScript runtime

V8

### Architecture

None

### Build type

Debug mode

### Device

Android emulator

### Device model

pixel 3a

### Acknowledgements

Yes

--- a/README.md
+++ b/README.md
@@ -210,7 +210,7 @@ npx expo install expo-build-properties
 ```
 
 3. (Optional) Android에서 난독화를 사용하실 경우, [Expo BuildProperties](https://docs.expo.dev/versions/latest/sdk/build-properties/) 를 이용해
-Proguard Rule을 [공식 문서]([공식 문서](https://developers.kakao.com/docs/latest/ko/android/getting-started#project-pro-guard))와 같이 설정해줍니다.
+Proguard Rule을 [공식 문서](https://developers.kakao.com/docs/latest/ko/android/getting-started#project-pro-guard)와 같이 설정해줍니다.
 
 ## Methods
 

--- a/README.md
+++ b/README.md
@@ -159,7 +159,7 @@ iOS의 경우 `yarn add @react-native-seoul/kakao-login` 이후 `npx pod-install
 
 6. (Optional) 앱 배포 시, 코드 축소, 난독화, 최적화를 하는 경우, 카카오 SDK를 제외해야 하기 때문에 **ProGuard 규칙 파일**에 다음 코드를 추가해주세요.
 
-[공식 문서]([공식 문서](https://developers.kakao.com/docs/latest/ko/android/getting-started#project-pro-guard))
+[공식 문서](https://developers.kakao.com/docs/latest/ko/android/getting-started#project-pro-guard)
 
 ```
 -keep class com.kakao.sdk.**.model.* { <fields>; }

--- a/README.md
+++ b/README.md
@@ -193,6 +193,8 @@ npx expo install expo-build-properties
         "@react-native-seoul/kakao-login",
         {
           "kakaoAppKey": "{{kakao api key}}",
+          "overrideKakaoSDKVersion": "2.11.2", // Optional, 
+          "kotlinVersion": "1.9.0" // #392
         }
       ],
       [

--- a/README.md
+++ b/README.md
@@ -21,7 +21,7 @@ React Native 카카오 로그인 라이브러리 입니다. `@react-native-seoul
 
 [카카오 로그인 Example Project](https://github.com/react-native-seoul/react-native-kakao-login/tree/main/KakaoLoginExample) 데모 화면
 
-<img src="https://user-images.githubusercontent.com/27461460/111674724-a7989f00-885f-11eb-9e51-d146757ca836.gif" width="200">
+<img alt="example" src="https://user-images.githubusercontent.com/27461460/111674724-a7989f00-885f-11eb-9e51-d146757ca836.gif" width="200">
 
 > 위 프로젝트는 `KakaoLoginExample` 폴더에서 확인 가능합니다.
 
@@ -59,31 +59,31 @@ iOS의 경우 `yarn add @react-native-seoul/kakao-login` 이후 `npx pod-install
 
 3. [공식문서 - 개발 프로젝트 설정](https://developers.kakao.com/docs/latest/ko/getting-started/sdk-ios) 을 참고하여 `info.plist`, `URL Types` 및 커스텀 스킴 추가 등 기타 필요한 세팅들을 프로젝트에 추가해줍니다. 아래`카카오 네이티브앱 아이디를 적어주세요` 문구를 잘 확인하시여 본인의 Kakao App Key로 변경해주세요.
 
-   ```diff
-    <key>CFBundleURLTypes</key>
-    <array>
-   + <dict>
-   +   <key>CFBundleTypeRole</key>
-   +   <string>Editor</string>
-   +   <key>CFBundleURLSchemes</key>
-   +   <array>
-   +     <string>kakao{카카오 네이티브앱 아이디를 적어주세요}</string>
-   +   </array>
-   + </dict>
-    </array>
-    <key>CFBundleVersion</key>
-    <string>1</string>
-   + <key>KAKAO_APP_KEY</key>
-   + <string>{카카오 네이티브앱 아이디를 적어주세요}</string>
-   + <key>KAKAO_APP_SCHEME</key> // 선택 사항 멀티 플랫폼 앱 구현 시에만 추가하면 됩니다
-   + <string>{카카오 앱 스킴을 적어주세요}</string> // 선택 사항 
-   + <key>LSApplicationQueriesSchemes</key>
-   + <array>
-   +   <string>kakaokompassauth</string>
-   +   <string>storykompassauth</string>
-   +   <string>kakaolink</string>
-   + </array>
-   ```
+```diff
+ <key>CFBundleURLTypes</key>
+ <array>
++ <dict>
++   <key>CFBundleTypeRole</key>
++   <string>Editor</string>
++   <key>CFBundleURLSchemes</key>
++   <array>
++     <string>kakao{카카오 네이티브앱 아이디를 적어주세요}</string>
++   </array>
++ </dict>
+ </array>
+ <key>CFBundleVersion</key>
+ <string>1</string>
++ <key>KAKAO_APP_KEY</key>
++ <string>{카카오 네이티브앱 아이디를 적어주세요}</string>
++ <key>KAKAO_APP_SCHEME</key> // 선택 사항 멀티 플랫폼 앱 구현 시에만 추가하면 됩니다
++ <string>{카카오 앱 스킴을 적어주세요}</string> // 선택 사항 
++ <key>LSApplicationQueriesSchemes</key>
++ <array>
++   <string>kakaokompassauth</string>
++   <string>storykompassauth</string>
++   <string>kakaolink</string>
++ </array>
+```
 
 4. `3.0.0` 버전부터는 swift 버전의 kakao sdk를 활용하므로 [Swift Bridging Header](https://stackoverflow.com/questions/31716413/xcode-not-automatically-creating-bridging-header)를 추가해야할 수 있습니다.
    <img width="800" alt="1" src="https://user-images.githubusercontent.com/27461460/111863065-8be6e300-899c-11eb-8ad8-6811e0bd0fbd.png">
@@ -159,16 +159,29 @@ iOS의 경우 `yarn add @react-native-seoul/kakao-login` 이후 `npx pod-install
 
 6. (Optional) 앱 배포 시, 코드 축소, 난독화, 최적화를 하는 경우, 카카오 SDK를 제외해야 하기 때문에 **ProGuard 규칙 파일**에 다음 코드를 추가해주세요.
 
-   ```
-   -keep class com.kakao.sdk.**.model.* { <fields>; }
-   -keep class * extends com.google.gson.TypeAdapter
-   ```
+[공식 문서]([공식 문서](https://developers.kakao.com/docs/latest/ko/android/getting-started#project-pro-guard))
+
+```
+-keep class com.kakao.sdk.**.model.* { <fields>; }
+-keep class * extends com.google.gson.TypeAdapter
+
+# https://github.com/square/okhttp/pull/6792
+-dontwarn org.bouncycastle.jsse.**
+-dontwarn org.conscrypt.*
+-dontwarn org.openjsse.**
+```
 
 7. Gradle 및 카카오 SDK의 버전을 변경해야 하는 경우, [android/gradle.properties](./android/gradle.properties) 에 있는 항목들을 확인하고, Android gradle의 root project의 ext에 `RNKakaoLogins_` 를 제외한 버전을 명시해주세요.
 
-#### EXPO (EAS Build only, SDK 41 이상)
+#### EXPO (Expo Go, Snack 사용 불가, Development Build(EAS, local build)만 가능)
 
-1. app.json 파일을 아래와 같이 수정합니다.
+1. Android의 Kakao SDK Maven Repository를 선언하기 위해 필요한 의존성을 추가합니다.
+
+```sh
+npx expo install expo-build-properties
+```
+
+2. app.json 파일을 아래와 같이 수정합니다.
 
 ```
 {
@@ -179,9 +192,15 @@ iOS의 경우 `yarn add @react-native-seoul/kakao-login` 이후 `npx pod-install
       [
         "@react-native-seoul/kakao-login",
         {
-          "kakaoAppKey": "", // 필수
-          "overrideKakaoSDKVersion": "2.9.0", // Optional, 
-          "kotlinVersion": "1.5.10" // Android Only, Optional, Expo 내부 패키지들과의 충돌이 있어 테스트 결과 1.5.10은 문제가 없었습니다. 지정 안하면 1.5.10으로 설정됩니다.
+          "kakaoAppKey": "{{kakao api key}}",
+        }
+      ],
+      [
+        "expo-build-properties",
+        {
+          "android": {
+            "extraMavenRepos": ["https://devrepo.kakao.com/nexus/content/groups/public/"]
+          }
         }
       ]
     ],
@@ -190,11 +209,8 @@ iOS의 경우 `yarn add @react-native-seoul/kakao-login` 이후 `npx pod-install
 } 
 ```
 
-SDK49 버전은 안드로이드 빌드 문제로 "kotlinVersion"을 `1.9.0`으로 설정해야합니다.
-
-2. EAS Build 이후 `expo start --dev-client`를 이용합니다.
-
-3. (Optional) Android에서 proguard rules 등을 적용하실 경우, [Expo BuildProperties](https://docs.expo.dev/versions/latest/sdk/build-properties/) 를 참고하세요
+3. (Optional) Android에서 난독화를 사용하실 경우, [Expo BuildProperties](https://docs.expo.dev/versions/latest/sdk/build-properties/) 를 이용해
+Proguard Rule을 [공식 문서]([공식 문서](https://developers.kakao.com/docs/latest/ko/android/getting-started#project-pro-guard))와 같이 설정해줍니다.
 
 ## Methods
 

